ql/src/test/queries/clientpositive/vector_struct_in2.q

```diff
@@ -0,0 +1,22 @@
+set hive.fetch.task.conversion=none;
+
+create table test (a string) partitioned by (y string, m string);
+insert into test values ('aa', 2022, 9);
+
+--original bug report
+select * from test where (y=year(date_sub('2022-09-11',4)) and m=month(date_sub('2022-09-11',4))) or (y=year(date_sub('2022-09-11',10)) and m=month(date_sub('2022-09-11',10)) );
+
+set hive.optimize.point.lookup=false;
```

Comment:
Why do you need to set these to false in the test? From the case description the original bug is not hit when either of the two is disabled, aren't we undermining the test in this way?

Reply:
The original repro case is a complex one so I wanted to break down the issue to simpler test cases.
The original case runs with point lookup optimization set to true, but it has to be switched off later for the simpler cases because it would optimize the problematic parts of the queries away before the vectorization happens.
I was thinking about creating multiple query files for the different scenarios but I decided not to do so to avoid the overhead of distinct test runs.